source/common/http/http2/codec_impl.cc

```diff
@@ -341,7 +341,14 @@ void ConnectionImpl::StreamImpl::encodeTrailersBase(const HeaderMap& trailers) {
 
 void ConnectionImpl::StreamImpl::encodeMetadata(const MetadataMapVector& metadata_map_vector) {
   parent_.updateActiveStreamsOnEncode(*this);
-  ASSERT(parent_.allow_metadata_);
+  if (!parent_.allow_metadata_ &&
```

Comment:
I would assume accepting metadata but not forwarding it is a bug or misconfiguration.  I think we definitely want to log here, your call if a trace style log, or an ENVOY_BUG.